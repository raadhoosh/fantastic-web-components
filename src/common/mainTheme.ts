import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    bottom: {
        fontsize: '1rem',
        border: '5px',
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
