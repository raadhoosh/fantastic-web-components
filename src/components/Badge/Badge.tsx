import styled from 'react-emotion';
import * as React from 'react';

interface IProps {
  /** 	The color of the component. It supports those theme colors that make sense for this component. */
  bgColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark';
  /** 	The color of the component. It supports those theme colors that make sense for this component. */
  borderColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark';
  /** The badge will be added relative to this node. */
  children?: any;
  /** Styles applied to the root element if color="#f0f" */
  color?: string;
  /** .It supports those theme */
  theme?: any;
}

const EmBadge = styled('span')(
  (props: (IProps)) => {
    const base_label = {
      display: 'flex',
      zIndex: 1,
      width: props.theme.badge.width,
      height: props.theme.badge.height,
      'flex-wrap': 'wrap',
      'align-items': 'center',
      'align-content': 'center',
      'border-radius': props.theme.badge.radius,
      'flex-direction': 'row',
      'justify-content': 'center',
      color: props.theme.badge.color,
      fontSize: props.theme.badge.fontSize,
      backgroundColor: props.theme.color.primary,
      border: props.theme.badge.borderSize,
      borderColor: props.theme.color.primary,
      borderStyle: props.theme.badge.borderStyle,
    };
    if (props.bgColor) {
      base_label.backgroundColor = props.theme.color[props.bgColor];
    }
    if (props.color) {
      base_label.color = props.color;
    }
    if (props.borderColor) {
      base_label.borderColor = props.theme.color[props.borderColor];
    } else {
      base_label.borderColor = 'transparent';
      base_label.border = '0';
      base_label.borderStyle = 'none';
    }
    return base_label;
  }
);

/**
 * The API documentation of the Badge React component. Learn more about the properties and the CSS customization points.
 */
const Badge = (props: IProps) => {

  return (
    <EmBadge {...props}> {props.children} </EmBadge>
  );
};

export default Badge;
