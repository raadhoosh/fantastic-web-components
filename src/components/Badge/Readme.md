
Badge example :

```jsx
<Badge bgColor="default" >5</Badge>
```

```jsx
<Badge bgColor="secondary" borderColor="secondary" >15</Badge>
```


```jsx
<Badge bgColor="primary" borderColor="primary" >230</Badge>
```

```jsx
<Badge bgColor="light" borderColor="dark" color="light" >23</Badge>
```