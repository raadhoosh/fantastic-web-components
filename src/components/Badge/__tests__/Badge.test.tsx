import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Badge from '../Badge';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const tree = renderer.create(<Badge theme={mainTheme} bgColor="primary">hi test</Badge>).toJSON();
    expect(tree).toHaveStyleRule('background-color', mainTheme.color.primary);
    expect(tree).toMatchSnapshot();
});
