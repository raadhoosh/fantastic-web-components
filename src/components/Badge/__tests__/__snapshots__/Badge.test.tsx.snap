// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<span
  className="css-1mljkwe"
>
   
  hi test
   
</span>
`;
